### PR TITLE
Add API notebook for makers

### PR DESCRIPTION
**Pull request by QRemy**
This PR add a makers notebook and remove duplication from rst page.

### CODE REVIEW

File: docs/tutorials/api/makers.ipynb [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+    "data using the `MapDatasetMaker`:\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+     "nbsphinx-thumbnail": {
+      "tooltip": "Data reduction : from observations to binned datasets"
+     },
```
Comment (QRemy) [inline docs/tutorials/api/makers.ipynb:104]: When I add the thumbnail tooltip in the metadata it break the notebook, what I am doing wrong here ?

Unreadable Notebook: /Users/qremy/Work/GitHub/gammapy/docs/tutorials/api/makers.ipynb NotJSONError('Notebook does not appear to be JSON: \'{\\n "cells": [\\n {\\n "cell_type": "m...')

Comment (QRemy) [reply docs/tutorials/api/makers.ipynb:104]: A bracket is missing, I will fix that here and in the dev doc page

File: docs/tutorials/api/makers.ipynb [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+    "dataset_empty = MapDataset.create(geom=geom)\n",
+    "print(dataset_empty)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The ``.create()`` method has\n",
+    "additional options, e.g. it is also possible to specify a true energy axis:"
```
Comment (AtreyeeS) [inline docs/tutorials/api/makers.ipynb:75]: Perhaps mention that this is the binning of the IRFs.
Also, it is possible to specify other quantities like `rad_axis`, `migra_axis` and `bins_irf`

File: docs/tutorials/api/makers.ipynb [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+     "tooltip": "Data reduction : from observations to binned datasets"
+    }},
+   "outputs": [],
+   "source": [
+    "#get observation\n",
+    "data_store = DataStore.from_dir(\"$GAMMAPY_DATA/hess-dl3-dr1\")\n",
+    "obs = data_store.get_observations([23592])[0]\n",
+    "\n",
+    "#fill dataset\n",
+    "maker = MapDatasetMaker()\n",
```
Comment (AtreyeeS) [inline docs/tutorials/api/makers.ipynb:112]: Mention about `background_oversampling` that can be passed on init

Comment (AtreyeeS) [inline docs/tutorials/api/makers.ipynb:112]: Mention about `background_oversampling` that can be passed on init

File: docs/tutorials/api/makers.ipynb [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+    "can be safely estimated as radially symmetric w.r.t. the pointing direction, one can apply the\n",
+    "reflected regions background technique.\n",
+    "This is implemented in the `~gammapy.makers.ReflectedRegionsBackgroundMaker` which transforms a\n",
+    "`SpectrumDataset` in a `SpectrumDatasetOnOff`. This technique is only used for 1D spectral\n",
+    "analysis.\n",
+    "\n",
+    "For more details and usage, see [reflected_background](https://docs.gammapy.org/dev/makers/reflected.html).\n",
+    "\n",
+    "\n",
+    "## Stacking of datasets\n",
```
Comment (AtreyeeS) [inline docs/tutorials/api/makers.ipynb:242]: I think we can omit stacking in this notebook and it goes in as independent documentation

**Review by AtreyeeS — COMMENTED** [review]
Thanks a lot @QRemy .
I left a couple of inline comments.

I think it will also be nice to add a small section for `SpectrumDatasetMaker`

File: docs/tutorials/api/makers.ipynb (hunk 2 above)
Comment (QRemy) [reply docs/tutorials/api/makers.ipynb:75]: - added : "it is also possible to specify a true energy axis **which defines the binning of the instrument response functions**:
- for the other options I will just show show them thought  a `help(MapDataset.create)` call, as a notebook don't have to repeat all the docstring

File: docs/tutorials/api/makers.ipynb (hunk 4 above)
Comment (QRemy) [reply docs/tutorials/api/makers.ipynb:242]: I removed the subsection but I still think that the text is relevant so I moved it as a note about the data reduction loop.

**Review by AtreyeeS — DISMISSED** [review]
Thanks! I left one small comment, its good from my side after this

File: docs/tutorials/api/makers.ipynb [hunk 5]
```diff
@@ -0,0 +1,412 @@
...
+    "    0.3, 10, nbin=31, name=\"energy_true\", unit=\"TeV\", interp=\"log\"\n",
+    ")\n",
+    "dataset_empty = MapDataset.create(geom=geom, energy_axis_true=energy_axis_true)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For the detail of the other options availables, you can always call the help:"
```
[the remark this answers is not in the document]
Comment (QRemy) [reply docs/tutorials/api/makers.ipynb:99]: Thanks for the suggestion, added

**Review by AtreyeeS — APPROVED** [review]

**Review by adonath — APPROVED** [review]
Thanks @QRemy! I think we should still spend some time to better define the separation of contents between RST pages and notebooks. It will probably go in a direction, where methods and theory is explained in the RST pages, while notebooks focus on code examples and API. However this we need to do in a consistent manner anyway...so specific comment for this PR.

**Pull request merged**